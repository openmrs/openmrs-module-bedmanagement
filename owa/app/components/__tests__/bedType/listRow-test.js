import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import BedTypeListRow from 'components/bedType/bedTypeList/bedTypelistRow';
import bedTypeFunctionsMock from 'components/__mocks__/bedTypeFunctions-mock';

const testProps = {
    bedTypeFunctions: bedTypeFunctionsMock
};

describe('BedTypeListRow', () => {
    it('Should render bed Type list properly', () => {
        const bedTypeListRow = mount(
            <table>
                <tbody>
                    <BedTypeListRow
                        bedType={testProps.bedTypeFunctions.getBedTypeByUuid('6f9fb240-0fd5-11e8-adb7-080027b38971')}
                        bedTypeFunctions={testProps.bedTypeFunctions}
                    />
                </tbody>
            </table>
        );

        expect(
            bedTypeListRow
                .find('td')
                .at(0)
                .text()
                .trim()
        ).toBe('luxury');
        expect(shallowToJson(bedTypeListRow)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnDeleteHandler = jest.spyOn(BedTypeListRow.prototype, 'deleteHandler');
        const spyOnEditHandler = jest.spyOn(BedTypeListRow.prototype, 'editHandler');
        const spyOnSetState = jest.spyOn(testProps.bedTypeFunctions, 'setState');
        const bedTypeList = mount(
            <table>
                <tbody>
                    <BedTypeListRow
                        bedType={testProps.bedTypeFunctions.getBedTypeByUuid('6f9fb240-0fd5-11e8-adb7-080027b38971')}
                        bedTypeFunctions={testProps.bedTypeFunctions}
                    />
                </tbody>
            </table>
        );

        bedTypeList.find('.fa-edit').simulate('click');
        expect(spyOnEditHandler).toHaveBeenCalled();
        expect(spyOnSetState).toHaveBeenCalledWith({
            activePage: 'addEdit',
            pageData: {
                operation: 'edit',
                bedTypeUuid: '6f9fb240-0fd5-11e8-adb7-080027b38971'
            }
        });

        bedTypeList.find('.fa-trash').simulate('click');
        expect(spyOnDeleteHandler).toHaveBeenCalled();
    });
});
