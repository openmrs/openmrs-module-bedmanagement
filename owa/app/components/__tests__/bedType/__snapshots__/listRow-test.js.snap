// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedTypeListRow Should render bed Type list properly 1`] = `
<table>
  <tbody>
    <BedTypeListRow
      bedType={
        Object {
          "description": "LXY",
          "displayName": "luxury bed",
          "name": "luxury",
          "uuid": "6f9fb240-0fd5-11e8-adb7-080027b38971",
        }
      }
      bedTypeFunctions={
        Object {
          "fetchBedTypes": [Function],
          "getBedTypeByUuid": [Function],
          "getBedTypeName": [Function],
          "getBedTypes": [Function],
          "notify": [Function],
          "setState": [Function],
        }
      }
    >
      <tr>
        <td>
          luxury
        </td>
        <td>
          luxury bed
        </td>
        <td>
          LXY
        </td>
        <td>
          <a
            href="javascript:void(0);"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="icon fa fa-edit"
            />
             Edit
          </a>
            |  
          <a
            href="javascript:void(0);"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="icon fa fa-trash"
            />
             Delete
          </a>
        </td>
      </tr>
    </BedTypeListRow>
  </tbody>
</table>
`;
